src/components/BlogList.jsx

```diff
@@ -1,22 +1,46 @@
 import BlogPost from "./BlogPost";
 import Pagination from "./Pagination";
-import React from "react";
+import React, {useState} from "react";
 import blogs from "../data/blogs.json";
 
 const PAGE_SIZES = [15, 25, 50, 100];
 
 function BlogList() {
-  const currentPaginationData = blogs.posts.slice(0, 15);
 
-  const updateRowsPerPage = () => {};
-  const updatePage = () => {};
+  // state that stores current page index
+  const [currentPage, setCurrentPage] = useState(1);
+
+  
+
+  // state that stores number of blogs to show per page
+  const [selectedPageSize, setSelectedPageSize] = useState(15);
+  const [sliceAmount, setSliceAmount] = useState(15);
+  const [sliceIndex, setSliceIndex] = useState(0);
+  
+  // updates row per page based on user selection
+  const updateRowsPerPage = (userPageSize) => {
+    console.log(userPageSize);
```

Comment:
It is always best to remove debugging statements after using them because the are not needed in production.